src/mg_decoration.cpp

```diff
@@ -294,7 +294,7 @@ bool DecoSimple::canPlaceDecoration(MMVManip *vm, v3s16 p)
 }
 
 
-size_t DecoSimple::generate(MMVManip *vm, PcgRandom *pr, v3s16 p)
+size_t DecoSimple::generate(Mapgen *mg, MMVManip *vm, PcgRandom *pr, v3s16 p)
```

Comment:
Wait, why are you adding the whole mapgen to the Schematic interface?  This violates the principle of least access.

I think you should be able to access the member `ndef` or `m_ndef` anyway as it should be inherited from ObjDef.